Dashboard.Storage/Azure/BuildUtil.cs

```diff
@@ -71,6 +71,30 @@ public List<BuildFailureEntity> GetTestCaseFailures(DateTimeOffset startDate, st
             return _buildFailureDateTable.ExecuteQuery(query).ToList();
         }
 
+        public List<string> GetViewNames(DateTimeOffset startDate)
```

Comment:
I agree, I think that would be extremely beneficial; while I made the query asynchronous to minimize the impact, it would at least allow the dropdown box on the page to get loaded much faster, particularly for a larger date range. 😄